peps/pep-0774.rst

```diff
@@ -0,0 +1,253 @@
+JIT itself but rather to determine whether the build-time dependency on LLVM is
+acceptable for JIT builds moving forward. If this PEP is rejected, we will
+proceed with the status quo, retaining the LLVM build-time requirement. While
+this dependency has served the JIT development process effectively thus far, it
+introduces setup complexity and additional challenges that this PEP seeks to
+alleviate.
+
+Motivation
+==========
+
+At the Python Core Developer Sprint that took place in September 2024, there was
+discussion about the next steps for the JIT - a related discussion also took
+place on `GitHub <https://github.com/python/cpython/issues/115869>`__. As part
+of that discussion, there was also a clear appetite for removing the LLVM
+requirement for JIT builds in preparation for shipping the JIT off by default in
+3.14. The consensus at the sprint was that it would be sufficient to provide
+pre-generated stencils for non-debug builds for Tier 1 platforms and that
+checking these files into the CPython repo would be adequate for the limited
+number of platforms (though more options have been explored; see Rejected
+Ideas).
+
+Currently, building CPython with `the JIT requires LLVM
+<https://github.com/python/cpython/tree/main/Tools/jit#installing-llvm>`__ as a
+build-time dependency. Despite not being exposed to end users, this dependency
+is suboptimal. Requiring LLVM adds a setup burden for developers and those who
+wish to build CPython with the JIT enabled. Depending on the operating system,
+the version of LLVM shipped with the OS may differ from that required by our JIT
+builds, which introduces additional complexity to troubleshoot and resolve. With
+few core developers currently contributing to and maintaining the JIT, we also
+want to make sure that the friction to work on JIT-related code is minimized as
+much as possible.
+
+With the proposed approach, hosting pre-compiled stencils for supported
+architectures can be generated in advance, stored in a central location, and
+automatically used during builds. This approach ensures reproducible builds,
+making the JIT a more stable and sustainable part of CPython’s future.
+
+Rationale
+=========
+
+This PEP proposes checking JIT stencils directly into the CPython repo as the
+best path forward for eliminating our build-time dependency on LLVM.
+
+This approach: 
+
+* Provides the best end-to-end experience for those looking to build CPython 
+  with the JIT
+* Lessens the barrier to entry for those looking to contribute to the JIT 
+* Ensures builds remain reproducible and consistent across platforms without 
+  relying on external infrastructure or download mechanisms 
+* Eliminates variability introduced by network conditions or potential
+  discrepancies between hosted files and the CPython repository state, and 
+* Subjects stencils to the same review processes we have for all other JIT-related
+  code
+
+However, this approach does result in a slight increase in overall
```

Comment:
Should this paragraph (or somewhere else) mention the current total size of the stencils?